### PR TITLE
Master published button fixes yti

### PR DESCRIPTION
**Pull request by tivisse**
ISSUE : https://www.odoo.com/web#id=644612&view_type=form&model=project.issue&action=963&active_id=681

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
I bet this breaks portal_sale ...

File: addons/payment_transfer/data/transfer.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <openerp>
-    <data noupdate="1">
+    <data>
```
Comment (tde-banana-odoo) [inline addons/payment_transfer/data/transfer.xml]: That means that at every update, the custom messages on the transfer acquirer will be reset to the default values ... does not seem a good idea to me :) .

File: addons/website/models/website.py [hunk 2]
```diff
@@ -814,7 +814,7 @@ def _website_url(self, cr, uid, ids, field_name, arg, context=None):
         return dict.fromkeys(ids, '#')
 
     def website_publish_button(self, cr, uid, ids, context=None):
-        if self.pool['res.users'].has_group(cr, uid, 'base.group_website_publisher'):
+        if self.pool['res.users'].has_group(cr, uid, 'base.group_website_publisher') and self.open_website_url(cr, uid, ids, context)['url'] != '#':
```
Comment (tde-banana-odoo) [inline addons/website/models/website.py]: Still not sure for this one.

File: addons/website_payment/__openerp__.py [hunk 3]
```diff
@@ -15,6 +15,7 @@
     'data': [
         'views/website_payment_templates.xml',
         'views/website_settings_payment.xml',
+        'data/transfer.xml',
     ],
-    'auto_install': False,
+    'auto_install': True,
```
Comment (tde-banana-odoo) [inline addons/website_payment/__openerp__.py]: Sure ?

File: addons/website/models/website.py (hunk 2 above)
Comment (tivisse) [inline addons/website/models/website.py]: Another solution is too add publish buttons in frontend and related urls to all elements which don't have any url, but not a good idea to me. This solution is generic, and @antonylesuisse was ok when I showed it to him.

File: addons/website_payment/__openerp__.py (hunk 3 above)
Comment (tivisse) [inline addons/website_payment/__openerp__.py]: Seems legit to me to be able to publish a payment method if 'website' and 'payment' are published without checking a option.

**Comment by antonylesuisse** [comment]
Merged in commits:
bd394f3c55cce19839e1e42a106d2b0342761243
ed4137495c64805e09265f6c2d6984675a2093c0
eed69de936df88b2d4a229660aeba37d944515e6
2cbb244f735f3f5567b3c8d279b59774c63adf4c

**Pull request closed without merging**